File: android2/app/src/main/java/com/xbot/xbot/XBotApplication.java
package com.xbot.xbot;

import android.app.Application;
import android.util.Log;

/**
 * Application entry point for the native Android shell.
 */
public class XBotApplication extends Application {
    private static final String TAG = "XBotApplication";
    private static XBotApplication instance;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        Log.i(TAG, "XBot native shell started");
    }

    public static XBotApplication getInstance() {
        return instance;
    }
}

File: xbot-native/app/src/main/java/com/xbot/android/XBotApplication.kt
package com.xbot.android

import android.app.Application
import android.os.StrictMode

/**
 * XBot 原生 Android 应用入口。
 *
 * 职责:
 * - 初始化全局单例(检测引擎、语音助手等)
 * - 配置 StrictMode(开发阶段)
 * - 持有 Application 级别的 Context 供 DI 使用
 */
class XBotApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        // 开发阶段启用 StrictMode 检测主线程 IO/网络
        if (BuildConfig.DEBUG) {
            StrictMode.setThreadPolicy(
                StrictMode.ThreadPolicy.Builder()
                    .detectAll()
                    .penaltyLog()
                    .build()
            )
        }
    }

    companion object {
        private const val TAG = "XBotApp"
    }
}
